pep-0387.txt

```diff
@@ -112,6 +112,28 @@ Basic policy for backwards compatibility
   platforms).
 
 
+Soft Deprecation
+================
+
+A soft deprecation can be used when using an API must no longer be used
+to write new code, but it remains safe to continue using it in existing
+code. The API remains documented and tested, but will not be developed
+further (no enhancement).
+
+The documentation of a soft deprecation should explain why the API
+should be avoided, and if possible propose a replacement.
+
+Soft deprecated functions must emit a SoftDeprecationWarning warning
+(new in Python 3.13) in the `Python Development Mode
+<https://docs.python.org/dev/library/devmode.html>`_ and if `Python is
+built in debug mode
+<https://docs.python.org/dev/using/configure.html#python-debug-build>`_:
+by default, this warning is not emitted.
```

Comment:
Can we be explicit that `-Werror` and similar "turn warnings into error" approaches *must not* make soft deprecation warnings errors?

Reply:
Running ``python3.13 -X dev -Werror`` will treat SoftDeprecationWarning as any other warnings as errors. If you want to treat them differently, you should dig into the warnings documentation to fine tune them.

For example, ``python3.13 -X dev -Werror -Walways::SoftDeprecationWarning`` will treat all warnings as error, but SoftDeprecationWarnings will just be displayed.

I don't think that a PEP is the right place to document warnings usage.

Reply:
Technically, a warning is an exception. Raising a warning means raising an exception: that's what ``-Werror`` does.

warnings.warn() says **issue a warning**: https://docs.python.org/dev/library/warnings.html#warnings.warn

I can replace "this warning is not not emitted" with "this warning is not issued". Would it be better like that?